public/app/core/selectors/navModel.ts

```diff
@@ -48,6 +33,26 @@ export const getNavModel = (navIndex: NavIndex, id: string, fallback?: NavModel,
   return getNotFoundModel();
 };
 
+function getSectionRoot(node: NavModelItem): NavModelItem {
+  if (!node.parentItem) {
+    return node;
+  }
+
+  const root = (node.parentItem = { ...node.parentItem });
+
+  if (root.children) {
+    root.children = root.children.map((item) => {
+      if (item.id === node.id) {
+        return { ...node, active: true };
```

Comment:
**Nit:** maybe by using `...item` here would make it easier to understand that we are just marking the nav item as "active" (even if the two are basically identical). What do you think?
```suggestion
        return { ...item, active: true };
```

Reply:
that breaks a test, I need to put node here as node has updated children